File: 1004/1004.go
/*
**@Time       : 2019-10-25
**@Author     : HuShengQuan
**@File       : 1004.go
**@Software   : vim
**@type       : pat practice
*/

package main

import (
    "fmt"
)

type Student struct {
    name string
    sno string
    grade uint
}

func main() {
    var n int
    fmt.Scanf("%d", &n)

    var s = make([]Student, n)
    max, min := 0, 0

    for i := 0; i < n; i++ {
        fmt.Scanf("%s%s%d", &s[i].name, &s[i].sno, &s[i].grade)
        if s[i].grade < s[min].grade {
            min = i
        } 
        if s[i].grade > s[max].grade {
            max = i
        }
    }

    fmt.Printf("%s %s\n%s %s", s[max].name, s[max].sno, s[min].name, s[min].sno)
}

File: 1004/1004.cpp
/***********************************************
 > @Time       : 2019-10-29
 > @Author     : hushengquan
 > @File       : 1004.cpp
 > @Software   : vim
 > @type       : pat practice
************************************************/
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    string max_name, max_sno, min_name, min_sno;
    int max = -1, min = 101;

    for (int i = 0; i < n; i++) {
        string name, sno;
        int grade;
        cin >> name >> sno >> grade;

        if (grade > max) {
            max = grade;
            max_name = name;
            max_sno = sno;
        }
        if (grade < min) {
            min = grade;
            min_name = name;
            min_sno = sno;
        }
    }

    cout << max_name << " " << max_sno << endl << min_name << " " << min_sno;

    return 0;
}
